src/test/java/FlightsReporterTests.groovy

```diff
@@ -0,0 +1,197 @@
+
+        assertEquals(FlightsReporter.getAllFlightsTo(flights, FlightsReporter.Airports.LONDON),
+                Arrays.asList(flights[2], flights[4], flights[7]))
+
+        assertNotEquals(FlightsReporter.getAllFlightsTo(flights, FlightsReporter.Airports.NEW_YORK),
+                Arrays.asList(flights[5], flights[6], flights[7]))
+    }
+
+    @Test
+    void testGetAllFlightsFrom() {
+        assert FlightsReporter.getAllFlightsFrom(flights, FlightsReporter.Airports.SOFIA) == [flights[1], flights[2], flights[3]]
+
+        assertEquals(FlightsReporter.getAllFlightsFrom(flights, FlightsReporter.Airports.NEW_YORK), [])
+
+        assertNotEquals(FlightsReporter.getAllFlightsFrom(flights, FlightsReporter.Airports.BERLIN), [])
+        assertEquals(FlightsReporter.getAllFlightsFrom(flights, FlightsReporter.Airports.PARIS),
+                [flights[0], flights[5], flights[8]])
+    }
+
+    @Test
+    void testPassengerAmount() {
+        def fl1 = FlightsReporter.getAllFlightsFrom(flights, FlightsReporter.Airports.NEW_YORK)
+        assert FlightsReporter.passengerAmount(fl1) == 0
+
+        assert FlightsReporter.passengerAmount(flights) == 1430
+        assertEquals(FlightsReporter.passengerAmountTo(flights, FlightsReporter.Airports.NEW_YORK), 360)
+        assertEquals(FlightsReporter.passengerAmountTo(flights, FlightsReporter.Airports.SOFIA), 230)
+        assertEquals(FlightsReporter.passengerAmountFrom(flights, FlightsReporter.Airports.SOFIA), 460)
+        assertEquals(FlightsReporter.passengerAmountFrom(flights, FlightsReporter.Airports.NEW_YORK), 0)
+    }
+
+    @Test
+    void testSortingByDepartureDate() {
+        def departureExpected = []
+        def fl1 = flights.collect()
+        departureExpected << flights[9] << flights[5] << flights[1] << flights[0] << flights[4] << flights[6]
+                << flights[2] << flights[3] << flights[8] << flights[7]
+        FlightsReporter.sortByDepartureDate(fl1)
+        assertEquals(fl1, departureExpected)
+    }
+
+    @Test
+    void testSortingByArrivalDate() {
+        def arrivalExpected = []
+        def fl2 = flights.collect()
+        arrivalExpected << flights[9] << flights[1] << flights[0] << flights[4] << flights[5] << flights[6]
+                << flights[2] << flights[3] << flights[8] << flights[7]
+        FlightsReporter.sortByArrivalDate(fl2)
+        assertEquals(fl2, arrivalExpected)
+    }
+
+    @Test
+    void testSortingByFlightLen() {
+        def flightLenExpected = []
+        def fl1 = flights.collect()
+        flightLenExpected << flights[0] << flights[7] << flights[4] << flights[1] << flights[2] << flights[9]
+                << flights[3] << flights[6] << flights[8] << flights[5]
+        FlightsReporter.sortByFlightLength(fl1)
+        assertEquals(fl1, flightLenExpected)
+    }
+
+    @Test
+    void testGetAllPlanesAboveCapacity() {
+        def expected = []
+        expected << flights[0] << flights[5] << flights[6]
```

Comment:
expected << flights[0] << flights[5] << flights[6] leads to 3 add() operators
Why not just
[flight[0], flight[5], flight[6]  ] ? It's way easier to read